CONTRIBUTING.md

```diff
@@ -0,0 +1,54 @@
+# Contributing to osbuild-composer
+
+First of all, thank you for taking the time to contribute to osbuild-composer. In this document you will find information that can help you with your contribution.
+
+### Running from sources
+
+We recommend using the latest stable Fedora for development but latest minor release of RHEL/CentOS 8 should work just fine as well. To run osbuild-composer from sources, follow these steps:
+
+```
+$ git clone https://github.com/osbuild/osbuild-composer.git
+$ cd osbuild-composer
+$ dnf group install 'RPM Development Tools' # Install rpmbuild
+$ dnf builddep golang-github-osbuild-composer.spec # Install build-time dependencies
+$ make rpm
+$ sudo dnf install rpmbuild/RPMS/x86_64/* -y
+$ sudo dnf install cockpit-composer -y
+$ sudo systemctl enable --now osbuild-composer.socket
+$ sudo systemctl enable --now cockpit.socket
+```
+
+You can now open https://localhost:9090 in your browser and check the "Image Builder" section.
+
+When developing the code, use `go` executable to build, run, and test you code [1]:
+
+```
+$ go test ./...
+$ go build ./...
+$ go run ./cmd/osbuild-pipeline/
+```
+
+### Testing
+
+See [this document](test/README.md) for more information about testing.
+
+### Planning the work
+
+In general we encourage you to first fill in an issue and discuss the feature you would like to work on before you start. This can prevent the scenario where you work on something we don't want to include in our code.
+
+That being said, you are of course welcome to implement an example of what you would like to achieve.
+
+### Creating a PR
+
+The commits in the PR should have these properties:
+
+* Preferably minimal and well documented
+  * Where minimal means: don't do unrelated changes even if the code is obviously wrong
+  * Well documented: both code and commit message
+  * The commit message should start with the module you work on, like: `weldr: `, or `rcm:`
+* The code should compile and the composer should start, so that we can run `git bisect` on it
+* All code should be covered with unit-tests
```

Comment:
-`with` +`by`, otherwise you ask for unit-tests all over the code ;)

Reply:
hehe, what a lovely difference :)